### PR TITLE
PIMOB-1191: Invalid string case of of pre-filled expiry date text.

### PR DESCRIPTION
**Pull request by ehab-al-cko**
[Jira ticket](https://checkout.atlassian.net/browse/PIMOB-1191?atlOrigin=eyJpIjoiNThlZTQ4ZTg5M2I5NGUzN2FlYmJmYWRlNTc5YzgxMTIiLCJwIjoiaiJ9
)
## Proposed changes

Fix: Invalid string case of of pre-filled expiry date text.
image

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift [hunk 1]
```diff
@@ -92,7 +90,14 @@ public final class ExpiryDateView: UIView {
    then validate full Expiry Date
    */
 
-  private func validateInput(_ textField: UITextField, currentDigit: Int, location: Int, replacementText: String) -> Bool {
+  func validateInput(_ textField: UITextField, location: Int, replacementText: String) -> Bool {
```
[the remark this answers is not in the document]
Comment (ehab-al-cko) [reply Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift:93]: It's already used in a test case

**Review by deepesh-vasthimal-cko — APPROVED** [review]

File: Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift:93]: It is used in a single unit test that you've updated in this PR to use it. That unit test was previously testing `validateInputChanges` so I don't understand why we need to expose this function to add a test that is working with the higher order.

The analogy I have is:
```swift

funa makePancakes() {
    // I don't know how to make pancakes so will limit to throwing pan in the air
    flipPancakesInPan()
}

private func flipPancakesInPan() {

}
```

Why would I need to expose `flipPancakesInPan` for unit test when I am concerned with outcome? So I want to know that the pancakes made are the way I expect them to be. I don't care how the flipping occurs.

Getting back to this code, you were already testing that given an input **x** with an addition of **y**, the behaviour was correct. Why do you want to test the private logic for how that worked originally? Important was it worked.

[the remark this answers is not in the document]
Comment (ehab-al-cko) [reply Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift]: Updated

**Review by alex-ioja-yang-cko — APPROVED** [review]

File: Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-ioja-yang-cko) [reply Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift]: Thank you. The goal is to maximise access control usage to make it easier at every level to gain certainty over functionality (available, how it works, etc). My concern is not applying it correctly will make it easier to introduce bugs and break things (for example tomorrow I may use the wrong validateInput method and lose functionality).

**Pull request merged**